server/sonar-web/src/main/webapp/WEB-INF/app/controllers/project_controller.rb

```diff
@@ -320,7 +320,7 @@ def update_version
       end
     end
 
-    redirect_to :action => 'history', :id => snapshot.root_project_id
+    redirect_to :action => 'history', :id => snapshot.root.id
```

Comment:
updating a version fails in webapp. Form is redirected on a wrong id. Are you sure that `snapshot.root` is the project but not the root snapshot ?